spikeinterface/toolkit/postprocessing/template_similarity.py

```diff
@@ -22,9 +26,17 @@ def compute_template_similarity(waveform_extractor, method='cosine_similarity'):
     s = templates.shape
     if method == 'cosine_similarity':
         templates_flat = templates.reshape(s[0], -1)
-        similarity = sklearn.metrics.pairwise.cosine_similarity(templates_flat)
+        if waveform_extractor_other is not None:
+            templates_other = waveform_extractor_other.get_all_templates()
+            s_other = templates_other.shape
+            templates_other_flat = templates_other.reshape(s_other[0], -1)
+            assert len(templates_flat[0]) == len(templates_other_flat[0]), ("Templates from second WaveformExtractor "
+                                                                            "don't have the correct shape!")
```

Comment:
I don't anderstand, we need that the 2 sorting have exactly the same number of units to compare then ?

Reply:
no, same number of channels